platform/linux-generic/odp_pool.c

```diff
@@ -135,6 +135,9 @@ static void cache_flush(pool_cache_t *cache, pool_t *pool)
 	ring_ptr_t *ring;
 	uint32_t mask;
 
+	if (!pool->ring)
```

Comment:
Ok, but if _odp_pool_create() fails halfway through, pool->ring may be set while the pool (including pool->ring->hdr ?) is not fully initialized.

I did not look in detail but the code seems a bit dubios in this regard.

Reply:
Added commit `linux-gen: pool: free ring shm if pool creation fails` in v3.